dds/idl/marshal_generator.cpp

```diff
@@ -3661,6 +3661,24 @@ namespace {
         "    uni._d(disc);\n"
         "    return true;\n"
         "  }\n"
+        "  if (disc == RTPS::PID_PROPERTY_LIST) {\n"
+        "    // support special case deserialization of DDS::PropertyQosPolicy\n"
+        "    ACE_Message_Block param(size);\n"
```

Comment:
It would be nice if we could avoid the allocation here. Is this a potential use case for the new peek / peek_helper functions on Serializer?